test/export/StructureDefinitionExporter.test.ts

```diff
@@ -1589,6 +1589,79 @@ describe('StructureDefinitionExporter', () => {
     expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
   });
 
+  it('should apply a FixedValue rule with a valid Canonical entity defined in FSH with a version', () => {
+    const profile = new Profile('MyObservation');
+    profile.parent = 'Observation';
+    const rule = new FixedValueRule('code.coding.system');
+    rule.fixedValue = new FshCanonical('VeryRealCodeSystem');
+    rule.fixedValue.useEntityVersion = true;
+    profile.rules.push(rule);
+
+    const realCodeSystem = new FshCodeSystem('VeryRealCodeSystem');
+    const caretRule = new CaretValueRule('');
+    caretRule.caretPath = 'version';
+    caretRule.value = '1.2.3';
+    realCodeSystem.rules.push(caretRule);
+    doc.codeSystems.set(realCodeSystem.name, realCodeSystem);
+
+    exporter.exportStructDef(profile);
+    const sd = pkg.profiles[0];
+    const fixedSystem = sd.findElement('Observation.code.coding.system');
+    expect(fixedSystem.patternUri).toEqual(
+      'http://hl7.org/fhir/us/minimal/CodeSystem/VeryRealCodeSystem|1.2.3'
+    );
+    expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
+  });
+
+  it('should use the config specified version when a version cannot be found for a Canonical entity', () => {
+    const profile = new Profile('MyObservation');
+    profile.parent = 'Observation';
+    const rule = new FixedValueRule('code.coding.system');
+    rule.fixedValue = new FshCanonical('VeryRealCodeSystem')
+      .withFile('Real.fsh')
+      .withLocation([1, 2, 3, 4]);
+    rule.fixedValue.useEntityVersion = true;
+    profile.rules.push(rule);
+
+    const realCodeSystem = new FshCodeSystem('VeryRealCodeSystem');
+    doc.codeSystems.set(realCodeSystem.name, realCodeSystem);
+
+    exporter.exportStructDef(profile);
+    const sd = pkg.profiles[0];
+    const fixedSystem = sd.findElement('Observation.code.coding.system');
+    // still should set the value
+    expect(fixedSystem.patternUri).toEqual(
+      'http://hl7.org/fhir/us/minimal/CodeSystem/VeryRealCodeSystem|1.0.0'
+    );
+  });
+
+  it('should log an error when no version can be found for a Canonical entity', () => {
+    delete fisher.tank.config.version;
```

Comment:
Could you restore this value after the test? Otherwise, other test cases that are added after this one that are expecting that value to be present wouldn't work as expected, and it could be hard to find the reason why.

Reply:
Good catch, done in https://github.com/FHIR/sushi/pull/536/commits/22a8a82ca7ced87af2cc888a00329617749309aa